### PR TITLE
Adding ruff to replace black/isort/flake8

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Infrastructure changes for replacing black/isort/flake8 with Ruff.

# Jira Ticket / Issue #

N/A

# Testing Instructions

N/A

# Dev Checklist:

N/A

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

- will update release notes when all the changes go in

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -126,16 +126,19 @@
...
     "scripts": {
-        "black": "black -v .",
-        "black_files": "black -v ",
+        "py-format": "ruff format .",
+        "py-format-files": "ruff format ",
+        "py-lint": "ruff check .",
+        "py-lint-files": "ruff check ",
+        "py-lint-files-exit-zero": "ruff check --exit-zero ",
+        "py-lint-fix": "ruff check --fix .",
+        "py-lint-fix-files": "ruff check --fix ",
```
Comment (ialarmedalien) [inline package.json:135]: these are all the fun new ruff commands (surprisingly!)

File: package.json [hunk 2]
```diff
@@ -166,9 +169,8 @@
             "npx prettier --ignore-unknown --write"
         ],
         "*.py": [
-            "npm run isort_files",
-            "npm run black_files",
-            "npm run flake8_files"
+            "npm run py-format-files",
+            "npm run py-lint-files-exit-zero"
```
Comment (ialarmedalien) [inline package.json:173]: this allows the linter to exit with status zero even if it picks up errors

File: package.json (hunk 1 above)
Comment (briehl) [reply package.json:135]: Shocking!

**Review by briehl — CHANGES_REQUESTED** [review]
Seems fine to me, but you could also change the bits of docs that reference using `flake8` here. `docs/git-workflow.md`, `docs/developer/running_tasks.md`.

**Review by briehl — APPROVED** [review]
Ok, I see the docs updated in the next PR. This one's fine. 👍

**Pull request merged**